pkg/cache/cache_test.go

```diff
@@ -0,0 +1,211 @@
+package cache_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/cache"
+	"github.com/openshift-psap/special-resource-operator/pkg/clients"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestCache(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cache Suite")
+}
+
+func resetGlobals() {
+	clients.Interface = nil
+
+	cache.Node.Count = 0xDEADBEEF
+	cache.Node.List = &unstructured.UnstructuredList{
+		Object: map[string]interface{}{},
+		Items:  []unstructured.Unstructured{},
+	}
+}
+
+var _ = Describe("Cache", func() {
+	Context("Nodes", func() {
+		BeforeEach(resetGlobals)
+		AfterEach(resetGlobals)
+
+		When("machingLabels=nil", func() {
```

Comment:
i think that removing When and using a simple look with It (as in previous comment), will make a code  a little bit clearer

Reply:
I reworked this file trying to reduce the number of `When` while still preserving the BDD spirit and the consistency. PTAL.